src/wallet/wallet.cpp

```diff
@@ -3262,7 +3262,8 @@ bool CWallet::SetAddressBook(const CTxDestination& address, const std::string& s
         LOCK(cs_wallet); // mapAddressBook
         std::map<CTxDestination, CAddressBookData>::iterator mi = mapAddressBook.find(address);
         fUpdated = mi != mapAddressBook.end();
-        mapAddressBook[address].name = strName;
+        if (!strName.empty()) /* update name only if requested */
```

Comment:
This is incorrect. If `mapAddressBook` doesn't already have an entry for `address`, an empty **has** to be created, or the resulting address will be considered change.

Reply:
Right, funny travis is green.

Reply:
It's green because there is no test case where strPurpose.isEmpty(). 

As far as #13087 concerned, I could think of:
* disallow adding key when it's already in the wallet
(so as to allow the user to recreate wallet and import the address again to change label) 
'importaddress' already disallow adding an existing address.

* extend existing rpc commands (import*key) to allow the user update label/purpose (possibly with an additional flag?). not sure how to allow the user to specify an explicit empty label.

* adding new rpc command for updating label/purpose (and remove)